presto-geospatial/src/main/java/com/facebook/presto/geospatial/BingTile.java

```diff
@@ -34,6 +37,39 @@ private BingTile(int x, int y, int zoomLevel)
         this.zoomLevel = zoomLevel;
     }
 
+    @Override
+    public boolean equals(Object other)
+    {
+        if (this == other) {
+            return true;
+        }
+
+        if (other == null || getClass() != other.getClass()) {
+            return false;
+        }
+
+        BingTile otherTile = (BingTile) other;
+        return this.x == otherTile.x
+            && this.y == otherTile.y
+            && this.zoomLevel == otherTile.zoomLevel;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(x, y, zoomLevel);
+    }
+
+    @Override
+    public String toString()
```

Comment:
I just tested to see whether this change is being reflected all the way to the presto-cli and I got another problem #9294.